### PR TITLE
Export segment group GUI

### PR DESCRIPTION
**Pull request by PaulHax**
Add a save button for each segment group
image

Clicking pops a dialog to select export format:
image

Saving downloads the segment group to the browsers download folder.

### CODE REVIEW

**Comment by floryst** [comment]
Nice addition! I would suggest having a dropdown selector for the file format instead of a radio.

File: src/components/SaveSegmentGroupDialog.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  if (fileName.value.trim().length === 0) {
+    return;
+  }
+  try {
+    saving.value = true;
+    const image = segmentGroupStore.dataIndex[props.id];
+    const serialized = await writeImage(fileFormat.value, image);
+    saveAs(new Blob([serialized]), `${fileName.value}.${fileFormat.value}`);
+    props.close();
+  } finally {
```
Comment (floryst) [inline src/components/SaveSegmentGroupDialog.vue:81]: If an error while saving occurs, does the UI behave/notify properly?

File: src/components/SaveSegmentGroupDialog.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  'dcm',
+  'hdf5',
+  'tif',
+  'mha',
+  'vtk',
+  'iwi.cbor',
+];
+
+const props = defineProps<{
+  close: () => void;
```
Comment (floryst) [inline src/components/SaveSegmentGroupDialog.vue:60]: Any reason for passing close as a prop vs emitting a close event? Just to be clear, this is fine as-is, but vue tends to prefer emitting events.

Comment (PaulHax) [reply src/components/SaveSegmentGroupDialog.vue]: Good tip.  React instincts. I changed it to an event.

**Pull request merged**